src/ripple/protocol/AnyPublicKey.h

```diff
@@ -102,9 +102,16 @@ class AnyPublicKey
 
 public:
     AnyPublicKey() = delete;
-    AnyPublicKey (AnyPublicKey const&) = delete;
     AnyPublicKey& operator= (AnyPublicKey const&) = delete;
 
+    AnyPublicKey (AnyPublicKey const& other)
+        : buffer_type(other.buffer_type::member.data(),
+            other.buffer_type::member.size())
+        , AnyPublicKeySlice (buffer_type::member.data(),
+            buffer_type::member.size())
+    {
+    }
```

Comment:
`unl.insertEphemeralKey (m->signingKey, iter->second.comment);`

It needs to copy the signing key.  Unless you want to risk storing an `AnyPublicKeySlice`…

Another way to save a memory allocation would be to have `AnyPublicKey` objects share the buffer (since they're inherently immutable values) but I don't think it's worth it.  This is not a performance or maintenance issue and we're grinding our own dust.  This inquiry has already resulted in a new error in the code (since fixed).